1981.1: Refactor application overview in administration

andrew8er opened the pull request:
### Short Description

In preparation of #1981 I refactored the application overview to use MUI components and adhere to the [design](https://www.figma.com/design/SDoGnXIjCCXpE1tAAJkRVc/entitlementcard?node-id=993-587&t=dNJn8iXuBM4mDzCv-1).

Almost all components are now using MUI components rather than Blueprint, except for some components used in the application filter bar and the application detail tree.

### Proposed Changes

- Update `eslint` dependencies.
- Update `@mui` dependencies.
- Update `date-fns` dependeny.
- Add a preliminary MUI theme for use throughout the application.
- Use MUI components and implement design in `/applications`

### Side Effects

None

### Testing

- Load test data with `backend: $ ./gradlew db-recreate`.
- Run backend and adminstration dev server.
- Visit http://localhost:3000/applications.
- Run tests.


### Resolved Issues

<!-- List all issues which should be closed when this PR is merged. -->

Partially addresses #1981

f1sh1918 commented on administration/src/bp-modules/applications/VerificationsQuickIndicator.tsx:
```diff
@@ -1,51 +1,57 @@
+  return (
+    <Container
+      ref={ref}
+      sx={{
```
❓Just a general question. Why do we not use the mui `styled` api here?
I mean if we want to style just a particular part of a complex component we have to need the particular props but i general i prefer the styled components style, since the readability is way better

andrew8er replied:
When making a one-off customization of a component, I generally use the `sx` prop. For components that get reused or I want to externalize, I would probably use the `styled()` approach. I tend to follow these guidelines:

https://mui.com/material-ui/customization/how-to-customize/#overriding-nested-component-styles

Though I'm a bit annoyed that there seems to be no API to combine both ways (make a component with a styled top-level element and provide e.g. custom `slotProps`) and that the `styled()` API are slightly different in functionality: https://mui.com/system/styled/#difference-with-the-sx-prop

f1sh1918 replied:
Hm my personal preference would be to use `styled()` if its possible even if the components are not reused because i find it better to read and it's clearer when to use what.
In all other cases where particular parts via props have to be styled for sure this way has to be used.
Thats also how we did it in other projects so far afaics.

But maybe others of the team have another opinion

f1sh1918 replied:
ok i'm fine to keep that for now.
yes sure we shouldn't mix styled apis

f1sh1918 commented on administration/src/i18n.ts:
```diff
@@ -32,6 +32,7 @@ export const loadTranslations = (): Translations => ({
     stores: deTranslations.stores,
     userSettings: deTranslations.userSettings,
     users: deTranslations.users,
+    component_Accordion: deTranslations.component_Accordion,
```
💭 hm currently we set namespaces for pages and we already, so i would tend to say this can be in `applicationOverview` 
If we really want to have a namespace for "base component" that we reuse in several places, we can even create a "baseComponents" namespace or put it in misc. 
I don't see an advantage in creating a namespace for each component.
What was your intend?

andrew8er replied:
Yes, I had the idea of one namespace per component for reusable components. I would like to refactor the file layout later after the MUI migration and wanted to introduce a `/shared` dir for components and other functionality at some point. 

We could use one namespace for all modules in (future) `/shared`, but I still would prefix component specific strings with e.g. `Accordion-` or something. Would you agree?

f1sh1918 commented on administration/src/App.tsx:
```diff
@@ -20,7 +20,9 @@ const App = (): ReactElement => (
         <AppApolloProvider>
           <LocalizationProvider dateAdapter={AdapterDateFns}>
-            <Router />
+            <ThemeProvider theme={theme}>
```
❌ The imports are missing for `ThemeProvider` and `theme` itself, so the app breaks

andrew8er replied:
I don't get it. This worked locally up until some point a few minutes ago. Should be already fixed.

f1sh1918 replied:
Just checked the example code of mui and the imports are defenitively needed :)

andrew8er replied:
I had the imports in there, but somehow they got deleted, probably in a rebase…

f1sh1918 commented on administration/src/bp-modules/applications/NoteDialogController.tsx:
```diff
@@ -67,7 +68,12 @@ const NoteDialogController = ({
     <>
       <Tooltip content={toolTipContent}>
-        <NoteButton onClick={() => onOpenNoteDialog(true)} intent='none' icon='annotation'>
+        <NoteButton
```
🙃 I'm not super happy with the styling of the button. Its gray and onHover white.
Not sure if this is intended in the design but seems weird to me since a hover should be highlighting.
But maybe thats just a personal preference

andrew8er replied:
I agree, did not test this thoroughly enough.

The MUI palette system is a bit lacking in view, too many common cases are not covered, for example, there is no configurable way for inverted panels, for example light text on dark background in a light theme. It can be fixed with module augmentation as shown in `src/types/mui-theme.d.ts` though.

I would like an official style guide here.

f1sh1918 reviewed (changes requested):
Looks really nice already. Great work 👍 
Theme provider imports are missing (see comments and ci pipeline), so the app is broken currently.

One small issue:
PDF Export the status bar is broken. I would tend to just hide it for print
<img width="465" alt="image" src="https://github.com/user-attachments/assets/28e77b14-d036-4811-86c2-3277cae1b007" />
And i added another minor button styling comment and namespaces

In general if you add TODOs please create an issue for it and refer to it :)

connium commented:
@andrew8er @f1sh1918 From my experience the best way is to tweak the theme settings and if something cannot be handled by theme overrides, we should ask ourselves if it is worth it to differ from the default design/behaviour (in most cases it is not).

f1sh1918 commented on administration/src/util/translations/de.json:
```diff
@@ -606,5 +606,9 @@
     "startDayLabel": "Startdatum"
+  },
+  "shared": {
+    "Accordion-showLess": "Weniger anzeigen",
```
🙃 i would like to keep the camel case conventions for namespaces
```suggestion
    "accordionShowLess": "Weniger anzeigen",
```

Same for showMore

bahaaTuffaha reviewed (approved):
Nice work ✅

🙃 nitpicking:
it looks different from the design
<img src="https://github.com/user-attachments/assets/31e6c01f-3ef7-4184-b25f-c829b010e784" width="300px" />
<img src="https://github.com/user-attachments/assets/f0340587-3414-4991-a5c4-af7151c27d2a" width="300px" />

Maybe for a future ticket: making application view responsive on mobile.🤔

bahaaTuffaha commented on administration/src/mui-modules/NonIdealState.tsx:
```diff
@@ -0,0 +1,20 @@
+import { Container, Stack, Typography, useTheme } from '@mui/material'
+import React, { ReactElement } from 'react'
+
+/* eslint-disable react/destructuring-assignment */
```
You can deconstruct it without adding an eslint rule:
`const NonIdealState = ({ title, icon, description }:{ title: string; icon: ReactElement; description: string }`

andrew8er commented:
> 🙃 nitpicking:
> it looks different from the design

You are right, I didn't see this dialogue in the design. But I think, it is more than nitpicking here. It looks like pretty out of place from all other design elements, and is inconsistent with other dialogues. I will check back with Toni and ask for clarification on this.

andrew8er commented:
@f1sh1918 @bahaaTuffaha I addressed all of your concerns and rebased the branch on top of `main` again.

As for the dialog styles, I kept it as it is for now, as I am not sure in what general direction the design will go. The biggest inconsistency is the position of the primary button, in one instance it is on the left, in the other on the right. The [Material2 design guide](https://m2.material.io/components/buttons#hierarchy-and-placement) states that the primary button goes to the right.

![Screenshot 2025-05-27 at 12 24 55](https://github.com/user-attachments/assets/a152c959-616d-4e9f-820f-e7e2829d04e3)
![Screenshot 2025-05-27 at 12 25 06](https://github.com/user-attachments/assets/6b5069de-f71b-4f79-8984-df4d213c0d8e)

f1sh1918 commented:
> @f1sh1918 @bahaaTuffaha I addressed all of your concerns and rebased the branch on top of `main` again.
> 
> As for the dialog styles, I kept it as it is for now, as I am not sure in what general direction the design will go. The biggest inconsistency is the position of the primary button, in one instance it is on the left, in the other on the right. The [Material2 design guide](https://m2.material.io/components/buttons#hierarchy-and-placement) states that the primary button goes to the right.
> 
> ![Screenshot 2025-05-27 at 12 24 55](https://github.com/user-attachments/assets/a152c959-616d-4e9f-820f-e7e2829d04e3)
> ![Screenshot 2025-05-27 at 12 25 06](https://github.com/user-attachments/assets/6b5069de-f71b-4f79-8984-df4d213c0d8e)
> 

@hauf-toni i think this should be reflected in the designs that primary buttons are on the right side
